### PR TITLE
Javascript javascript2 week2

### PR DESCRIPTION
**Pull request by andrii1**

### CODE REVIEW

File: javascript/javascript2/week2/hyf-bay/main.js [hunk 1]
```diff
@@ -1,17 +1,56 @@
...
+    const sortedProducts = products.sort((a, b) => a.rating - b.rating);
+    renderProducts(sortedProducts);
+  } else {
+    const sortedProducts = products.sort((a, b) => b.rating - a.rating);
+    renderProducts(sortedProducts);
+  }
+
+  clickCountRating++;
+}
+
```
Comment (Vahablotfi) [inline javascript/javascript2/week2/hyf-bay/main.js:53]: Well done ,This is a very cool feature to sort from low to high and other way around by double clicking. I would encourage you to add styling or text to makes it understandable for user.

File: javascript/javascript2/week2/movies.js [hunk 2]
```diff
@@ -44630,13 +44630,11 @@ console.log(moviesWithKeywords.length);
```
Comment (Vahablotfi) [inline javascript/javascript2/week2/movies.js:44630]: you could alternatively use spread operator to add tags to movie objects.(for exercise  4)

**Review by Vahablotfi — COMMENTED** [review]
Very good. It seems that you have a very good knowledge of JavaScript. I just had two recommends nothing important. Well done

**Comment by dpfernandes** [comment]
@27ManishaPatel please leave a review here :)

File: javascript/javascript2/week2/hyf-bay/main.js [hunk 3]
```diff
@@ -1,17 +1,56 @@
...
+productsCount.innerText = `${products.length} available`;
 searchInput.addEventListener("input", filterProducts);
-maximumPriceInput.addEventListener("input", filterProducts);
+maximumPriceInput.addEventListener("input", filterProductsByPrice);
+buttonSortPrice.addEventListener("click", sortProductsByPrice);
+buttonSortRating.addEventListener("click", sortProductsByRating);
 function filterProducts(event) {
   const filteredProducts = products.filter((product) =>
     product.name.toLowerCase().includes(event.target.value)
   );
```
Comment (27ManishaPatel) [inline javascript/javascript2/week2/hyf-bay/main.js:19]: Isn't it  good idea to use toLowerCase() method with input value as well..
like  product.name.toLowerCase().includes(event.target.value.toLowerCase()).

File: javascript/javascript2/week2/hyf-bay/main.js (hunk 1 above)
Comment (27ManishaPatel) [inline javascript/javascript2/week2/hyf-bay/main.js:53]: good idea with click count

File: javascript/javascript2/week2/hyf-bay/main.js [hunk 4]
```diff
@@ -22,13 +61,22 @@ function renderProducts(products) {
...
+    addToCartIcon.classList.add("fa-cart-arrow-down");
     list.appendChild(listItem);
     list.appendChild(name);
     list.appendChild(price);
     list.appendChild(rating);
+    list.appendChild(addToCartLink);
+    addToCartLink.appendChild(addToCartIcon);
+    addToCartLink.href = "#";
   }
 }
```
Comment (27ManishaPatel) [inline javascript/javascript2/week2/hyf-bay/main.js:81]: good job with extra feature " add to cart"

File: javascript/javascript2/week2/movies.js (hunk 2 above)
Comment (27ManishaPatel) [inline javascript/javascript2/week2/movies.js:44630]: I would like to write comment before each exercise to make it readable

Comment (27ManishaPatel) [reply javascript/javascript2/week2/movies.js:44630]: @Vahablotfi It's nice if you share some good reference for **spread operator** as I would like to learn it too

**Review by 27ManishaPatel — APPROVED** [review]
Nice work !

File: javascript/javascript2/week2/movies.js (hunk 2 above)
Comment (andrii1) [reply javascript/javascript2/week2/movies.js:44630]: good suggestion

File: javascript/javascript2/week2/hyf-bay/main.js (hunk 3 above)
Comment (andrii1) [reply javascript/javascript2/week2/hyf-bay/main.js:19]: ah good suggestion!

**Pull request merged**